internal/controller/phases.go

```diff
@@ -562,6 +572,65 @@ func getProvider(provider operatorv1.GenericProvider, defaultVersion string) clu
 	return *clusterctlProvider
 }
 
+func (p *phaseReconciler) getCustomProviders(ctx context.Context) ([]operatorv1.GenericProvider, error) {
+	customProviders := []operatorv1.GenericProvider{}
+	currProviderName := p.provider.GetName()
+	currProviderType := p.provider.GetType()
+
+	providerLists := []client.ObjectList{
+		&operatorv1.CoreProviderList{},
+		&operatorv1.BootstrapProviderList{},
+		&operatorv1.ControlPlaneProviderList{},
+		&operatorv1.InfrastructureProviderList{},
+		&operatorv1.AddonProviderList{},
+		&operatorv1.IPAMProviderList{},
+		&operatorv1.RuntimeExtensionProviderList{},
+	}
```

Comment:
The provider lists are already collected inside the `operatorv1.ProviderLists` slice. It would be nice to use it.